src/main/java/net/atomique/ksar/ui/NaturalComparator.java

```diff
@@ -17,7 +17,7 @@
   public final static Comparator<String> INSTANCE = new NaturalComparator();
   public final static Comparator<String> NULLS_FIRST = Comparator.nullsFirst(INSTANCE);
 
-  private final static Pattern WORD_PATTERN = Pattern.compile("\\s*+([^0-9\\s]|\\d+)");
+  private final static Pattern WORD_PATTERN = Pattern.compile("\\s*+([^0-9\\s]+|\\d+)");
```

Comment:
```suggestion
  private final static Pattern WORD_PATTERN = Pattern.compile("\\s*+((?>[^0-9\\s]++|\\d++))");
```

Reply:
I never used [possessive quantifiers](https://www.regular-expressions.info/possessive.html) ++
It looks like you're using Atomic Grouping and ++ which is not required ? .. Shouldn't it be "either or" ?

Reply:
I guess they are independent.
Atomic Grouping is like "if the first alternative **matched** then don't even **try others**".
possessive is like "if the char **matched** don't even try **backtrack**"

Reply:
just mentioned this because of ..
> Using Atomic Grouping Instead of Possessive Quantifiers
> Technically, possessive quantifiers are a notational convenience to place an atomic group around a single quantifier. All regex flavors that support possessive quantifiers also support atomic grouping. But not all regex flavors that support atomic grouping support possessive quantifiers. With those flavors, you can achieve the exact same results using an atomic group.

Reply:
Ah, you mean `((?>[^0-9\\s]++|\\d++))` and `((?>[^0-9\\s]+|\\d+))` are effectively the same thing? Could be. I did not know that.